src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -304,8 +304,10 @@ private void SolutionEvents_AfterClosing()
             }
         }
 
-        public void CleanCache()
+        public async Task CleanCacheAsync()
         {
+            await SolutionManager.CleanAsync();
```

Comment:
It can't go into Job since it's a non shared MEF component and only exposes restore api publicly.

I was ok having it in SolutionManager itself but if we just want to limit it to query methods then it will better to keep it in Worker itself.